### PR TITLE
[fix](move-memtable) fix nullptr and token num in TabletStream::close

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Check `nullptr` in `TabletStream::close`. And fix `_num_threads` in `LoadStreamMgr`.

```
#0  pthread_sigmask (how=2, newmask=<optimized out>, oldmask=0x0) at pthread_sigmask.c:48
#1  0x00007fe448ccd71e in PosixSignals::chained_handler(int, siginfo*, void*) [clone .part.0] () from /usr/lib/jvm/java-17-openjdk-amd64/lib/server/libjvm.so
#2  0x00007fe448cce206 in JVM_handle_linux_signal () from /usr/lib/jvm/java-17-openjdk-amd64/lib/server/libjvm.so
#3  <signal handler called>
#4  doris::ThreadPoolToken::wait (this=0x0) at /mnt/disk1/chenkaijie/doris/be/src/util/threadpool.cpp:161
#5  0x00005606566d9f68 in doris::TabletStream::close()::$_0::operator()() const (this=0x7fdc7cc72a00) at /mnt/disk1/chenkaijie/doris/be/src/runtime/load_stream.cpp:229
#6  std::__invoke_impl<void, doris::TabletStream::close()::$_0&>(std::__invoke_other, doris::TabletStream::close()::$_0&) (__f=...)
    at /mnt/disk1/chenkaijie/env/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:61
#7  std::__invoke_r<void, doris::TabletStream::close()::$_0&>(doris::TabletStream::close()::$_0&) (__fn=...)
    at /mnt/disk1/chenkaijie/env/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:111
#8  std::_Function_handler<void (), doris::TabletStream::close()::$_0>::_M_invoke(std::_Any_data const&) (__functor=...)
    at /mnt/disk1/chenkaijie/env/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_function.h:290
#9  0x00005606566c866e in std::function<void ()>::operator()() const (this=0x7fe036130a38)
    at /mnt/disk1/chenkaijie/env/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_function.h:591
#10 doris::WorkThreadPool<false>::work_thread (this=0x7fe4292d0010, thread_id=<optimized out>) at /mnt/disk1/chenkaijie/doris/be/src/util/work_thread_pool.hpp:158
#11 0x0000560662ae29cf in execute_native_thread_routine ()
#12 0x00007fe44d74c609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#13 0x00007fe44d9f9133 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95 
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — COMMENTED** [review]
LGTM

**Comment by kaijchen** [comment]
run buildall

File: be/src/runtime/load_stream.cpp [hunk 1]
```diff
@@ -223,12 +223,13 @@ Status TabletStream::close() {
...
-    auto wait_func = [this, &mu, &cv] {
+    auto wait_func = [this, &cv] {
         signal::set_signal_task_id(_load_id);
         for (auto& token : _flush_tokens) {
-            token->wait();
+            if (token != nullptr) {
+                token->wait();
+            }
         }
-        std::lock_guard<bthread::Mutex> lock(mu);
```
Comment (liaoxin01) [inline be/src/runtime/load_stream.cpp:231]: why remove this lock？

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: be/src/runtime/load_stream.cpp [hunk 2]
```diff
@@ -226,7 +226,9 @@ Status TabletStream::close() {
     auto wait_func = [this, &mu, &cv] {
         signal::set_signal_task_id(_load_id);
         for (auto& token : _flush_tokens) {
-            token->wait();
+            if (token != nullptr) {
```
Comment (liaoxin01) [inline be/src/runtime/load_stream.cpp:229]: When is token nullptr? If it is nullptr, an error should be reported？

**Comment by kaijchen** [comment]
run buildall

**Pull request closed without merging**